snap/validate_test.go

```diff
@@ -78,6 +78,33 @@ func (s *ValidateSuite) TestValidateEpoch(c *C) {
 	}
 }
 
+func (s *ValidateSuite) TestValidateHook(c *C) {
+	validHooks := []*HookInfo{
+		&HookInfo{Name: "a"},
+		&HookInfo{Name: "aaa"},
+		&HookInfo{Name: "a-a"},
+		&HookInfo{Name: "aa-a"},
+		&HookInfo{Name: "a-aa"},
+		&HookInfo{Name: "a-b-c"},
+	}
+	for _, hook := range validHooks {
+		err := ValidateHook(hook)
+		c.Assert(err, IsNil)
+	}
+	invalidHooks := []*HookInfo{
+		&HookInfo{Name: ""},
+		&HookInfo{Name: "a a"},
+		&HookInfo{Name: "0"},
+		&HookInfo{Name: "123"},
+		&HookInfo{Name: "abc0"},
+		&HookInfo{Name: "日本語"},
```

Comment:
Adding to Gustavos point, maybe we should add here `a--a` and `-a` and `a-` once the validateHook gets udated.